Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3738,3 +3771,88 @@ extension BrowserViewController: UNUserNotificationCenterDelegate {
         completionHandler()
     }
 }
+
+extension BrowserViewController: PlaylistHelperDelegate {
+    func showPlaylistAlert(_ alertController: UIAlertController) {
+        self.present(alertController, animated: true)
+    }
+    
+    func showPlaylistToast(info: PlaylistInfo, itemState: PlaylistItemAddedState) {
+        guard Preferences.Playlist.showToastForAdd.value,
+              let selectedTab = tabManager.selectedTab,
+              selectedTab.url?.isPlaylistSupportedSiteURL == true else {
+            return
+        }
+        
+        if let toast = playlistToast {
+            toast.item = info
+            return
+        }
+        
+        // Item requires the user to choose whether or not to add it to playlists
+        switch itemState {
+        case .pendingUserAction:
+            let toast = PlaylistToast(item: info, state: .itemPendingUserAction) { [weak self] buttonPressed in
+                guard let self = self, let info = self.playlistToast?.item else { return }
+                if buttonPressed {
+                    // Update playlist with new items..
+                    PlaylistItem.addItem(info, cachedData: nil) {
+                        PlaylistManager.shared.autoDownload(item: info)
```

Comment:
You cant because in that case you will have circular dependency between Data and Client. I didnt try but I guess you will even get a compilation error.